tests-e2e/core/snapshotenvironmentbinding_test.go

```diff
@@ -125,6 +126,74 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler E2E tests", func() {
 			Expect(gitOpsDeploymentSecond.OwnerReferences[0].UID).To(Equal(binding.UID))
 		})
 
+		// Verifies a SnapshotEnvironmentBinding's status component deployment condition is set correctly when the
+		// deployment of the components succeeds.
+		It("updates the binding's status component deployment condition when the deployment of the components succeeds.", func() {
+
+			By("creating binding cr and update the status field, because it is not updated when creating the object.")
+
+			k8sClient, err := fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			binding := buildSnapshotEnvironmentBindingResource("appa-staging-binding", "new-demo-app", "staging", "my-snapshot", 3, []string{"component-a", "component-b"})
+			err = k8s.Create(&binding, k8sClient)
+			Expect(err).To(Succeed())
+
+			// Update Status field
+			err = buildAndUpdateBindingStatus(binding.Spec.Components,
+				"https://github.com/redhat-appstudio/managed-gitops", "main", "fdhyqtw",
+				[]string{"resources/test-data/sample-gitops-repository/components/componentA/overlays/staging", "resources/test-data/sample-gitops-repository/components/componentB/overlays/staging"}, &binding)
+			Expect(err).To(BeNil())
+
+			By("checking the status component deployment condition is true")
+			Eventually(binding, "3m", "1s").Should(bindingFixture.HaveComponentDeploymentCondition(
+				metav1.Condition{
+					Type:    appstudiocontroller.ComponentDeploymentConditionAllComponentsDeployed,
+					Status:  metav1.ConditionTrue,
+					Reason:  appstudiocontroller.ComponentDeploymentConditionCommitsSynced,
+					Message: "2 of 2 components deployed",
+				}))
+
+			By("updating the bindings status field to force an out-of-sync component")
+			// Update Status field
+			err = buildAndUpdateBindingStatus(binding.Spec.Components,
+				"https://github.com/redhat-appstudio/managed-gitops", "main", "fdhyqtw",
+				[]string{"resources/test-data/sample-gitops-repository/components/componentA/overlays/staging", "resources/test-data/sample-gitops-repository/components/componentC/overlays/staging"}, &binding)
+			Expect(err).To(BeNil())
+
+			By("checking the status component deployment condition is false")
+			Eventually(binding, "3m", "1s").Should(bindingFixture.HaveComponentDeploymentCondition(
+				metav1.Condition{
+					Type:    appstudiocontroller.ComponentDeploymentConditionAllComponentsDeployed,
+					Status:  metav1.ConditionFalse,
+					Reason:  appstudiocontroller.ComponentDeploymentConditionCommitsUnsynced,
+					Message: "1 of 2 components deployed",
+				}))
+			Consistently(binding, "1m", "1s").Should(bindingFixture.HaveComponentDeploymentCondition(
```

Comment:
Is it necessary to use `Consistently` after `Eventually`?

Reply:
The `Consistently` check is there in this case to ensure the condition stays false.

Reply:
Okay, now I get it.